cmd/telemeter-rhel-server/main.go

```diff
@@ -0,0 +1,395 @@
+		telemeter_http.HealthRoutes(mux)
+		external.Mount("/", mux)
+
+		// v2 routes
+		{
+			v2ForwardClient := forwardClient
+
+			receiver, err := receive.NewHandler(o.Logger, o.ForwardURL, v2ForwardClient, prometheus.DefaultRegisterer, "tenantid", o.Whitelist, o.ElideLabels)
+			if err != nil {
+				level.Error(o.Logger).Log("msg", "could not initialize receive handler", "err", err)
+			}
+
+			external.Handle("/metrics/v1/receive", server.InstrumentedHandler("receive", http.HandlerFunc(receiver.Receive)))
+		}
+
+		externalPathJSON, _ := json.MarshalIndent(Paths{Paths: []string{"/", "/authorize", "/upload", "/healthz", "/healthz/ready", "/metrics/v1/receive"}}, "", "  ")
+
+		external.Get("/", func(w http.ResponseWriter, req *http.Request) {
+			w.Header().Add("Content-Type", "application/json")
+			if _, err := w.Write(externalPathJSON); err != nil {
+				level.Error(o.Logger).Log("msg", "could not write external paths", "err", err)
+			}
+		})
+
+		s := &http.Server{
+			Handler: otelhttp.NewHandler(external, "external", otelhttp.WithTracerProvider(tp)),
+			ErrorLog: stdlog.New(
+				&filteredHTTP2ErrorWriter{
+					out:               os.Stderr,
+					toDebugLogFilters: logFilter,
+					logger:            o.Logger,
+				},
+				"",
+				0),
+		}
+
+		// Run the external server.
+		g.Add(func() error {
+			if len(o.TLSCertificatePath) > 0 {
+				cert, err := tls.LoadX509KeyPair(o.TLSCertificatePath, o.TLSKeyPath)
+				if err != nil {
+					return err
+				}
+
+				caCert, err := ioutil.ReadFile(o.TLSCACertificatePath)
+				if err != nil {
+					return err
+				}
+				caCertPool := x509.NewCertPool()
+				caCertPool.AppendCertsFromPEM(caCert)
+
+				tlsConfig := &tls.Config{
+					Certificates: []tls.Certificate{cert},
+					ClientAuth:   tls.RequireAndVerifyClientCert,
+					ClientCAs:    caCertPool,
+				}
+
+				externalTLSListener := tls.NewListener(externalListener, tlsConfig)
+				if err := s.Serve(externalTLSListener); err != nil && err != http.ErrServerClosed {
+					level.Error(o.Logger).Log("msg", "external HTTPS server exited", "err", err)
+					return err
+				}
+
+			} else {
+				if err := s.Serve(externalListener); err != nil && err != http.ErrServerClosed {
+					level.Error(o.Logger).Log("msg", "external HTTP server exited", "err", err)
+					return err
+				}
+			}
+			return nil
+		}, func(error) {
+			_ = s.Shutdown(context.TODO())
+			externalListener.Close()
+
+			// Close clients in order to check for leaks properly.
+			forwardClient.CloseIdleConnections()
+			if c, ok := ctx.Value(oauth2.HTTPClient).(*http.Client); ok {
+				c.CloseIdleConnections()
+			}
+		})
+	}
+
+	// Kill all when caller requests to.
+	gctx, gcancel := context.WithCancel(ctx)
+	g.Add(func() error {
+		<-gctx.Done()
+		return gctx.Err()
+	}, func(err error) {
+		gcancel()
+	})
+
+	level.Info(o.Logger).Log("msg", "starting telemeter-server", "external", externalListener.Addr().String(), "internal", internalListener.Addr().String())
```

Comment:
lets remove telemeter here to avoid confusion going forward

Reply:
done